src/plugins/rgbcolor/rgbcolor.c

```diff
@@ -0,0 +1,379 @@
+					  { "powderblue", 0xb0e0e6ff },
+					  { "purple", 0x800080ff },
+					  { "red", 0xff0000ff },
+					  { "rosybrown", 0xbc8f8fff },
+					  { "royalblue", 0x4169e1ff },
+					  { "saddlebrown", 0x8b4513ff },
+					  { "salmon", 0xfa8072ff },
+					  { "sandybrown", 0xf4a460ff },
+					  { "seagreen", 0x2e8b57ff },
+					  { "seashell", 0xfff5eeff },
+					  { "sienna", 0xa0522dff },
+					  { "silver", 0xc0c0c0ff },
+					  { "skyblue", 0x87ceebff },
+					  { "slateblue", 0x6a5acdff },
+					  { "slategray", 0x708090ff },
+					  { "slategrey", 0x708090ff },
+					  { "snow", 0xfffafaff },
+					  { "springgreen", 0x00ff7fff },
+					  { "steelblue", 0x4682b4ff },
+					  { "tan", 0xd2b48cff },
+					  { "teal", 0x008080ff },
+					  { "thistle", 0xd8bfd8ff },
+					  { "tomato", 0xff6347ff },
+					  { "turquoise", 0x40e0d0ff },
+					  { "violet", 0xee82eeff },
+					  { "wheat", 0xf5deb3ff },
+					  { "white", 0xffffffff },
+					  { "whitesmoke", 0xf5f5f5ff },
+					  { "yellow", 0xffff00ff },
+					  { "yellowgreen", 0x9acd32ff },
+					  { NULL, 0 } };
+
+typedef enum
+{
+	COLOR_INVALID,
+	HEX_THREE,
+	HEX_FOUR,
+	HEX_SIX,
+	HEX_EIGHT,
+	NAMED_COLOR
+} ColorVariant;
+
+static const NamedColor * findNamedColor (const char * colorName)
+{
+	const NamedColor * cur = &NamedColors[0];
+	while (cur->name != NULL)
+	{
+		if (strcmp (cur->name, colorName) == 0)
+		{
+			return cur;
+		}
+		++cur;
+	}
+	return NULL;
+}
+
+static ColorVariant is_valid_key (Key * key, Key * parentKey)
+{
+	const Key * meta = keyGetMeta (key, "check/rgbcolor");
+	if (!meta) return 1;
+	const char * value = keyString (key);
+	const char * regexString = "^#([0-9a-fA-F]{3,4}|[0-9a-fA-F]{6}|[0-9a-fA-F]{8})$";
+
+	regex_t regex;
+	regmatch_t offsets;
+	int compile_failure = regcomp (&regex, regexString, REG_NOSUB | REG_EXTENDED | REG_NEWLINE);
+	if (compile_failure) return -1;
+	// regexec returns 0 on a successful match
+	int match = !(regexec (&regex, value, 0, &offsets, 0));
+	regfree (&regex);
+
+	const NamedColor * namedColor = findNamedColor (value);
+
+	if (namedColor != NULL)
+	{
+		return NAMED_COLOR;
+	}
+
+	if (!match && namedColor == NULL)
+	{
+		ELEKTRA_SET_ERRORF (214, parentKey, "Key %s with value %s is neither a valid hex formatted color nor a named color.",
+				    keyName (key), keyString (key));
+		return COLOR_INVALID;
+	}
+
+	int len = strlen (value);
+	switch (len)
+	{
+	case 4:
+		return HEX_THREE;
+	case 5:
+		return HEX_FOUR;
+	case 7:
+		return HEX_SIX;
+	case 9:
+		return HEX_EIGHT;
+	default:
+		return COLOR_INVALID;
+	}
+}
+
+static void elektraColorSetInteger (Key * key, kdb_unsigned_long_t c)
+{
+	char colorStr[11];
+	snprintf (colorStr, 11, "%u", c);
+
+	ELEKTRA_LOG_DEBUG ("Set %s to integer %s", keyName (key), colorStr);
+	keySetString (key, colorStr);
+	keySetMeta (key, "type", "unsigned_long");
+}
+
+static char * elektraColorExpand (const char * str, ColorVariant colVar)
+{
+	// Expand #abcd to #aabbccdd or #abc to #aabbccff
```

Comment:
The documentation for the function should use doxygen. So something like:

```c
/**
 * Expand #abcd to #aabbccdd or #abc to #aabbccff
 */
static char * elektraColorExpand (const char * str, ColorVariant colVar)
```

Also, I think "Expands all color variants to the full RRGGBBAA variant." better describes what happens here. Since the returned string is allocated with `elektraMalloc`, the documentation should also state that it has to be freed with `elektraFree`.

The `malloc` could also be avoided by passing in a `char *` similarly to `elektraWriteArrayNumber`.